src/main/java/org/folio/helper/PurchaseOrderLineHelper.java

```diff
@@ -204,48 +204,44 @@ public Future<CompositePoLine> createPoLine(CompositePoLine compPOL, JsonObject
    * @param compOrder associated {@link CompositePurchaseOrder} object
    * @return completable future which might hold {@link CompositePoLine} on success or an exception if any issue happens
    */
-  public Future<CompositePoLine> createPoLineWithOrder(CompositePoLine compPoLine, CompositePurchaseOrder compOrder,
-      RequestContext requestContext) {
+  public Future<CompositePoLine> createPoLineWithOrder(CompositePoLine compPoLine, CompositePurchaseOrder compOrder, RequestContext requestContext) {
     // The id is required because sub-objects are being created first
-    if (isEmpty(compPoLine.getId())) {
+    if (StringUtils.isEmpty(compPoLine.getId())) {
       compPoLine.setId(UUID.randomUUID().toString());
     }
     compPoLine.setPurchaseOrderId(compOrder.getId());
     updateEstimatedPrice(compPoLine);
     PoLineCommonUtil.updateLocationsQuantity(compPoLine.getLocations());
 
-    JsonObject line = mapFrom(compPoLine);
+    var line = convertToPoLine(compPoLine);
     List<Future<Void>> subObjFuts = new ArrayList<>();
-
     subObjFuts.add(createAlerts(compPoLine, line, requestContext));
     subObjFuts.add(createReportingCodes(compPoLine, line, requestContext));
 
     return GenericCompositeFuture.join(new ArrayList<>(subObjFuts))
       .compose(v -> generateLineNumber(compOrder, requestContext))
-      .map(lineNumber -> line.put(PO_LINE_NUMBER, lineNumber))
+      .map(line::withPoLineNumber)
       .compose(v -> purchaseOrderLineService.updateSearchLocations(compPoLine, requestContext))
-      .map(v -> line.put(SEARCH_LOCATION_IDS, compPoLine.getSearchLocationIds()))
+      .map(v -> line.withSearchLocationIds(compPoLine.getSearchLocationIds()))
       .compose(v -> createPoLineSummary(compPoLine, line, requestContext));
   }
 
   public Future<Void> setTenantDefaultCreateInventoryValues(CompositePoLine compPOL, JsonObject tenantConfiguration) {
-    Promise<JsonObject> promise = Promise.promise();
+    if (!isCreateInventoryNull(compPOL)) {
+      return Future.succeededFuture();
+    }
 
-    if (isCreateInventoryNull(compPOL)) {
-      if (tenantConfiguration != null && !tenantConfiguration.isEmpty() &&
-                      StringUtils.isNotEmpty(tenantConfiguration.getString(CREATE_INVENTORY))) {
-        promise.complete(new JsonObject(tenantConfiguration.getString(CREATE_INVENTORY)));
-      } else {
-        promise.complete(new JsonObject());
-      }
-      return promise.future()
-        .map(jsonConfig -> {
-          updateCreateInventory(compPOL, jsonConfig);
-          return null;
-        });
+    Promise<JsonObject> promise = Promise.promise();
+    if (tenantConfiguration != null && !tenantConfiguration.isEmpty() && !StringUtils.isEmpty(tenantConfiguration.getString(CREATE_INVENTORY))) {
+      promise.complete(new JsonObject(tenantConfiguration.getString(CREATE_INVENTORY)));
     } else {
-      return Future.succeededFuture();
+      promise.complete(new JsonObject());
```

Comment:
if you started refacroting here, we can improve even more by deleting promise usage